client/castApp/src/caster.c

```diff
@@ -129,6 +129,16 @@ void casterShutdown(caster_t *self)
     self->shutdown = 1;
     epicsMutexUnlock(self->lock);
 
+    if(self->extra_envs) {
+        epicsMutexMustLock(self->lock);
+        int i;
+        for(i = 0; self->extra_envs[i]; i++) {
+            free(self->extra_envs[i]);
+        }
+        free(self->extra_envs);
+        epicsMutexUnlock(self->lock);
```

Comment:
Do we need this extra lock/unlock in here? Can't we just move the `epicsMutexUnlock`from above to below this conditional block?

Reply:
Yes we could do that. I thought there might be some benefit to be explicit about which parts need locking. But that's probably not useful

Reply:
resolved with https://github.com/ChannelFinder/recsync/pull/57/commits/7baa5f8d08a1386656d876fa2a81519f34ce5aad